pyeudiw/satosa/backend.py

```diff
@@ -299,13 +292,16 @@ def redirect_endpoint(self, context, *args):
             if isinstance(decrypted_data["vp_token"], str)
             else decrypted_data["vp_token"]
         )
+        
+        # TODO: The issuer JWK must be selected from a dictinoary where the keys are the issuers and
+        # the values are the jwk connected to that issuer
+        issuer_jwk = JWK(self.config["metadata_jwks"][0])
```

Comment:
this MUST be taken from the kid value not just the first value [0]

please use this:
https://github.com/italia/eudi-wallet-it-python/blob/dev/pyeudiw/federation/trust_chain_validator.py#L20

please move the previous function in pyeudiw.jwk.utils and rename that function to `find_jwk_by_kid`

Reply:
The kid is the one present in the vp_token header?

Reply:
yes

Reply:
Resolved in [adfc12f](https://github.com/italia/eudi-wallet-it-python/pull/89/commits/adfc12fd4c6933fca8ea0c9d2655712c46970107)